### PR TITLE
extract attribute from properties function

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
close https://github.com/vesoft-inc/nebula-ent/issues/1157

## Checklist:
Tests:
- [x] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Add tests?

File: src/graph/visitor/PropertyTrackerVisitor.cpp [hunk 1]
```diff
@@ -174,27 +171,78 @@ void PropertyTrackerVisitor::visit(AttributeExpression *expr) {
...
+      auto funName = funCallExpr->name();
+      std::transform(funName.begin(), funName.end(), funName.begin(), ::tolower);
+      if (funName != "properties") {
+        break;
+      }
+      auto *argExpr = funCallExpr->args()->args().front();
+      auto kind = argExpr->kind();
+      switch (kind) {
+        case Expression::Kind::kInputProperty: {
+          //  match (v) return properties(v).name
```
Comment (jievince) [inline src/graph/visitor/PropertyTrackerVisitor.cpp:214]: What if `match ()-[e]->() return properties(e).name`?

File: src/graph/visitor/PropertyTrackerVisitor.cpp [hunk 2]
```diff
@@ -174,27 +171,78 @@ void PropertyTrackerVisitor::visit(AttributeExpression *expr) {
...
+      }
+      auto *argExpr = funCallExpr->args()->args().front();
+      auto kind = argExpr->kind();
+      switch (kind) {
+        case Expression::Kind::kVarProperty:
+        case Expression::Kind::kInputProperty: {
+          //  match (v) return properties(v).name
+          auto *inputPropExpr = static_cast<InputPropertyExpression *>(argExpr);
+          auto &aliasName = inputPropExpr->prop();
+          propsUsed_.insertVertexProp(aliasName, unKnowType_, propName);
```
Comment (jievince) [inline src/graph/visitor/PropertyTrackerVisitor.cpp:217]: How to known `v` is a vertex or edge?

Comment (nevermore3) [reply src/graph/visitor/PropertyTrackerVisitor.cpp:217]: edge is only subscriptExpression or listComprehensionExpression,  can be distinguished by the expression type

Comment (jievince) [reply src/graph/visitor/PropertyTrackerVisitor.cpp:217]: Is it invalid to use `properties(edge).name`?

Comment (nevermore3) [reply src/graph/visitor/PropertyTrackerVisitor.cpp:217]: edge is a KEYWORD, parser error

**Review by jievince — APPROVED** [review]

File: tests/tck/features/optimizer/PrunePropertiesRule.feature [hunk 3]
```diff
@@ -3,12 +3,10 @@
 # This source code is licensed under Apache 2.0 License.
 Feature: Prune Properties rule
 
-  Background:
```
Comment (Shylock-Hg) [inline tests/tck/features/optimizer/PrunePropertiesRule.feature:6]: Why change this?

**Review by Shylock-Hg — APPROVED** [review]

**Pull request merged**